### ISSUE TITLE
Bug: Duplicate diffs for location mismatch when running tests with Mocha

### ISSUE THREAD

**Issue by jaymarvelz** [eslint/eslint#19996]
### Environment

Node version: v24.3.0
npm version: 11.4.2
Local ESLint version: 9.32.0
Global ESLint version: 9.32.0
Operating System: Win11

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```js
<!-- Paste your configuration here -->
```
</details>

You can reproduce with a tiny custom rule and a RuleTester test run under Mocha that intentionally mismatches the expected error line:

```js
const { RuleTester } = require("eslint");

const rule = {
	meta: {
		messages: { always: "Always report" },
	},
	create(context) {
		return {
			Program(node) {
				context.report({
					node,
					messageId: "always",
					loc: {
						start: { line: 2, column: 0 },
						end: { line: 2, column: 1 },
					},
				});
			},
		};
	},
};

const ruleTester = new RuleTester();

ruleTester.run("always-report", rule, {
	valid: [],
	invalid: [
		{
			code: "\nfoo()",
			errors: [{ messageId: "always", line: 0 }],
		},
	],
});

```

### What did you expect to happen?

A single, clear diff (Mocha’s), or a single concise assertion message without duplicated diffs.

### What actually happened?

Two diffs are printed:
```
AssertionError [ERR_ASSERTION]: Actual error location does not match expected error location.
+ actual - expected

  {
+   line: 2
-   line: 0
  }

      + expected - actual

       {
      -  "line": 2
      +  "line": 0
       }
```

### Link to Minimal Reproducible Example

n/a

### Participation

- [x] I am willing to submit a pull request for this issue.

### Additional comments

- The first diff seems to come from Node’s assert.deepStrictEqual.
- The second diff is Mocha’s rendering based on actual/expected.

A change in RuleTester’s assertion strategy (e.g., custom comparison without deepStrictEqual-generated diff) would make failures much easier to read.

**Comment by snitin315** [comment]
I don't think this is a bug in ESLint code, this seems to be the behavior of `assert.deepStrictEqual` with `mocha`. Refer following example:

<img width="917" height="743" alt="Image" src="https://github.com/user-attachments/assets/72b2b514-eaad-4cae-873c-d01c0bd25fa5" />

<img width="931" height="873" alt="Image" src="https://github.com/user-attachments/assets/e9a985bd-9ccb-4f18-8fb1-5731fd2bf7d1" />

**Comment by crimsonjay0** [comment]
Yes — and that’s why I’m suggesting that RuleTester use a custom comparison instead of Node’s assert.deepStrictEqual, so we can avoid the extra deepStrictEqual-generated diff.

**Comment by crimsonjay0** [comment]
@mdjermanovic, since you previously reviewed #19904 which introduced this issue, your input on this would be valuable.

**Comment by mdjermanovic** [comment]
I think the current behavior is fine since RuleTester can be used with other testing frameworks that possibly don't display the diff, or even directly (without any testing framework), in which cases including the diff in the error message seems useful. When a testing framework that does display a diff is used, an extra diff is, in my opinion, not as important issue to justify removing the diff from error messages and thus not show the diff in the mentioned use cases.

> [#19904](https://github.com/eslint/eslint/pull/19904) which introduced this issue

Duplicate display of expected and actual values existed before #19904, maybe it was just less obvious. For example, this was the output with the example from the original post:

```
AssertionError [ERR_ASSERTION]: Error line should be 0

2 !== 0

      + expected - actual

      -2
      +0
```

The `2 !== 0` part comes from `assert.strictEqual`, while the + expected / - actual part is from Mocha.
